### PR TITLE
Linting updates for the analyzer given the new flake8 rules

### PR DESCRIPTION
**Pull request by omri374**
Going over the entire analyzer code base (anonymizer not yet included as it is WIP). 
Fixing:
1. Docstrings for all public methods
2. Type hints for input and output
3. General formatting and some variable names updates

### CODE REVIEW

File: presidio-analyzer/presidio_analyzer/__init__.py [hunk 1]
```diff
@@ -9,14 +11,15 @@
 from presidio_analyzer.recognizer_registry import RecognizerRegistry
 from presidio_analyzer.analyzer_engine import AnalyzerEngine
 
+logging.getLogger("presidio-analyzer").addHandler(logging.NullHandler())
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/__init__.py:14]: Updating logger based on ADR. This is just one line so putting it in this PR.

File: presidio-analyzer/presidio_analyzer/analyzer_engine.py [hunk 2]
```diff
@@ -2,41 +2,44 @@
...
 from presidio_analyzer import (
-    PresidioLogger,
     RecognizerRegistry,
     RecognizerResult,
     EntityRecognizer,
+    PresidioLogger,
 )
 from presidio_analyzer.app_tracer import AppTracer
-from presidio_analyzer.nlp_engine import NLP_ENGINES
+from presidio_analyzer.nlp_engine import NLP_ENGINES, NlpEngine
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/analyzer_engine.py:11]: Some imports are required due to added type hints

File: presidio-analyzer/presidio_analyzer/nlp_engine/nlp_artifacts.py [hunk 3]
```diff
@@ -1,18 +1,40 @@
...
     """
 
-    def __init__(self, entities, tokens, tokens_indices, lemmas, nlp_engine, language):
+    def __init__(
+        self,
+        entities: List[str],
+        tokens: Doc,
+        tokens_indices: List[int],
+        lemmas: List[str],
+        nlp_engine,  # noqa ANN001
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/nlp_engine/nlp_artifacts.py:20]: Cannot reference here due to circular import.

File: presidio-analyzer/setup.cfg [hunk 4]
```diff
@@ -1,6 +1,10 @@
...
+    __pycache__,
+    build,
+    dist,
+    tests
 docstring-convention = numpy
-per-file-ignores =
-    tests/*: D
-extend-ignore = D100,D101,D102,D104,D105,D103,D200,D202,D205,D404,D208,D210,E251,D400,D401,D403,F401,W503,N803
\ No newline at end of file
+extend-ignore = E203 D107 D100 D104 D202 ANN101 ANN102 ANN204 ANN203
```
Comment (SharonHart) [inline presidio-analyzer/setup.cfg:10]: D107 is already ignored in numpy convention

**Review by SharonHart — APPROVED** [review]

File: presidio-analyzer/presidio_analyzer/app_tracer.py [hunk 5]
```diff
@@ -3,10 +3,12 @@
 
 class AppTracer:
     """
-    This class provides the ability to log/trace the system's decisions,
-    such as which modules were used for detection,
+    Allow the logging/tracing the system's decisions.
```
[the remark this answers is not in the document]
Comment (omri374) [reply presidio-analyzer/presidio_analyzer/app_tracer.py]: Fixed

**Review by SharonHart — APPROVED** [review]
LGTM!

**Review by navalev — APPROVED** [review]

**Pull request merged**